### PR TITLE
Sniper gains an integrated laze in its gun.

### PR DESCRIPTION
**Pull request by Rohesie**
## Why It's Good For The Game

The philosophy design for the sniper is the long-range support and debuffer, more than death from afar. The idea is to reduce off-screen deaths as much as possible, giving some functions now as part of the Scout to the Sniper, and making the Scout more suited for closer range engagements.

## Changelog
:cl:
add: Sniper gains an integrated laze in its gun. Use CTRL+click, same as a regular tactical binocular.
refactor: Refactors the Laser Targeting into a Target Marker.
/:cl:

### CODE REVIEW

**Comment by Surrealaser** [comment]
I like it.

**Comment by Borisvanmemes** [comment]
Maybe give it a mode that makes it fire the railgun without command needing to to push the button for it?

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -96,6 +99,8 @@
 
 /obj/item/weapon/gun/rifle/sniper/M42A/Destroy()
 	laser_off()
+	qdel(integrated_laze)
```
Comment (Ghommie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: `QDEL_NULL(integrated_laze)`, and remove the line below.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -39,6 +40,8 @@
 	select_gamemode_skin(type, list(MAP_ICE_COLONY = "s_m42a"))
 	. = ..()
 	LT = image("icon" = 'icons/obj/items/projectiles.dmi',"icon_state" = "sniper_laser", "layer" =-LASER_LAYER)
+	integrated_laze = new
```
Comment (Ghommie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: `integrated_laze = new(src)`, and remove the line below.

File: code/_onclick/handle_click.dm [hunk 3]
```diff
@@ -8,6 +8,13 @@
 		return TRUE
 	return FALSE
 
+/obj/item/weapon/gun/rifle/sniper/M42A/handle_click(mob/living/user, atom/A, list/mods)
+	if (mods["ctrl"])
+		integrated_laze.acquire_target(A, user)
+		return TRUE
+	else
```
Comment (Ghommie) [inline code/_onclick/handle_click.dm]: no need for `else` here.

**Comment by Rohesie** [comment]
That would kill the overwatch functions, Boris, but in the long run after we give overwatch nicer tools it might be somthing to consider.

**Pull request merged**